src/components/ListComposable/utils/parseDataWithChildren.ts

```diff
@@ -0,0 +1,39 @@
+import type {ListItemType, ParsedState} from '../types';
+
+import {getGroupedId, parseGroupedId} from './groupedIdUtils';
+
+export function parseDataWithChildren<T>(items: ListItemType<T>[]): ParsedState<T> {
+    console.time('parseDataWithChildren');
+    const result = {
+        byId: {},
+        groupsState: {},
+        itemsState: {},
+    } as ParsedState<T>;
+
+    const traverseItems = (item: ListItemType<T>, index: number, parentId?: string) => {
+        const groupedId = getGroupedId(index, parentId);
```

Comment:
Suggestion: use unique id instead of index

Reply:
current implementation supports user uniq ids or fallback on build in custom id builder what reflects the current nest structure of list like `1-2-1`